src/tests/test_fastod.cpp

```diff
@@ -0,0 +1,509 @@
+#include <algorithm>
+#include <filesystem>
+#include <memory>
+#include <string>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/od/fastod/canonical_od.h"
+#include "algorithms/od/fastod/fastod.h"
+#include "config/names.h"
+#include "config/time_limit/type.h"
+
+namespace {
+
+namespace fs = std::filesystem;
+constexpr auto input_data_path = "input_data/od_norm_data/";
+
+inline size_t combine_hashes(size_t first, size_t second) {
+    size_t wave = second + 2654435769UL + (first << 6) + (first >> 2);
+    return first ^ wave;
+}
+
+size_t RunFastod(std::string const& dataset_name, std::string const& subfolder) {
+    fs::path dataset_path = fs::path(input_data_path) / subfolder / dataset_name;
+
+    algos::fastod::DataFrame data = algos::fastod::DataFrame::FromCsv(dataset_path);
+    algos::fastod::Fastod fastod(std::move(data));
+
+    std::vector<std::string> string_ods = fastod.DiscoverAsStrings();
+    std::sort(std::begin(string_ods), std::end(string_ods));
+
+    std::size_t result_hash = 0;
+
+    for (std::string const& od : string_ods) {
+        size_t od_hash = std::hash<std::string>{}(od);
+        result_hash = combine_hashes(result_hash, od_hash);
+    }
+
+    return result_hash;
+}
+
+}  // namespace
+
+class FastodTest : public ::testing::Test {
+protected:
+    static std::unique_ptr<algos::fastod::Fastod> CreateAlgorithmInstance(
+            config::InputTable table, config::TimeLimitSecondsType time_limit_seconds = 0u) {
+        using namespace config::names;
+
+        algos::StdParamsMap params{{kTable, table}, {kTimeLimitSeconds, time_limit_seconds}};
```

Comment:
it's better to pass here kCsvConfig, because AlgoFactory will create InputTable from this config.

Reply:
Did it.